### PR TITLE
Hl without distant

### PR DESCRIPTION
**Pull request by Uhutown**
All should work with the Exit Signal and also the signalinteroperability.

### CODE REVIEW

**Comment by Jeronimo97** [comment]
- Sh1 on Hl without distant doesn't work with signalbox
- Some HL signalling issues with signal box system (see at test server)

File: src/main/java/eu/gir/girsignals/signalbox/config/HVSignalConfig.java [hunk 1]
```diff
@@ -90,6 +104,37 @@ public void change(final int speed, final SignalTileEnity current, final SignalT
                 values.put(SignalHV.HPHOME, HPHome.HP1);
                 values.put(SignalHV.STOPSIGNAL, HP.HP1);
             }
+
+            final ArrayList<HL> stophlmain = Lists.newArrayList(HL.HP0, HL.HP0_ALTERNATE_RED,
+                    HL.HL_SHUNTING, HL.HL_ZS1);
+            final ArrayList<HLExit> stophlexit = Lists.newArrayList(HLExit.HP0,
+                    HLExit.HP0_ALTERNATE_RED, HLExit.HL_SHUNTING, HLExit.HL_ZS1);
+            final ArrayList<HL> hl40main = Lists.newArrayList(HL.HL2_3, HL.HL5_6, HL.HL8_9,
+                    HL.HL11_12);
```
Comment (MrTroble) [inline src/main/java/eu/gir/girsignals/signalbox/config/HVSignalConfig.java:113]: Same goes here maybe make class variable

File: src/main/java/eu/gir/girsignals/signalbox/config/HVSignalConfig.java [hunk 2]
```diff
@@ -90,6 +104,37 @@ public void change(final int speed, final SignalTileEnity current, final SignalT
...
+                    || next.getProperty(SignalHL.EXITSIGNAL).filter(d -> stophlexit.contains(d))
+                            .isPresent();
+            final boolean hlmain40 = next.getProperty(SignalHL.STOPSIGNAL)
+                    .filter(c -> hl40main.contains(c)).isPresent()
+                    || next.getProperty(SignalHL.EXITSIGNAL).filter(HLExit.HL2_3::equals)
+                            .isPresent();
+            final ArrayList<KS> stopks = Lists.newArrayList(KS.HP0, KS.KS_SHUNTING, KS.KS_ZS1,
+                    KS.KS_ZS7);
+            final ArrayList<KSMain> stopksmain = Lists.newArrayList(KSMain.HP0, KSMain.KS_SHUNTING,
+                    KSMain.KS_ZS1, KSMain.KS_ZS7);
```
Comment (MrTroble) [inline src/main/java/eu/gir/girsignals/signalbox/config/HVSignalConfig.java:125]: Make static class variable

File: src/main/java/eu/gir/girsignals/signalbox/config/KSSignalConfig.java [hunk 3]
```diff
@@ -41,7 +52,49 @@ public void change(final int speed, final SignalTileEnity current, final SignalT
...
             final boolean stop = next.getProperty(SignalKS.MAINSIGNAL).filter(KSMain.HP0::equals)
                     .isPresent() || opt.filter(KS.HP0::equals).isPresent();
-            if (stop) {
+
+            final ArrayList<HL> stophlmain = Lists.newArrayList(HL.HP0, HL.HP0_ALTERNATE_RED,
+                    HL.HL_SHUNTING, HL.HL_ZS1);
+            final ArrayList<HLExit> stophlexit = Lists.newArrayList(HLExit.HP0,
+                    HLExit.HP0_ALTERNATE_RED, HLExit.HL_SHUNTING, HLExit.HL_ZS1);
+            final ArrayList<HL> hl40main = Lists.newArrayList(HL.HL2_3, HL.HL5_6, HL.HL8_9,
+                    HL.HL11_12);
```
Comment (MrTroble) [inline src/main/java/eu/gir/girsignals/signalbox/config/KSSignalConfig.java:61]: Make static

**Review by MrTroble — CHANGES_REQUESTED** [review]

**Review by Jeronimo97 — APPROVED** [review]

**Review by MrTroble — APPROVED** [review]

**Pull request merged**